core/src/ephemeral_server/mod.rs

```diff
@@ -187,9 +199,45 @@ impl EphemeralServer {
         Err(anyhow!("Failed connecting to test server after 5 seconds"))
     }
 
-    /// Shutdown the server (i.e. kill the child process).
+    /// Shutdown the server (i.e. kill the child process). This does not attempt
+    /// a kill if the child process appears completed, but such a check is not
+    /// atomic so a kill could still fail as completed if completed just before
+    /// kill.
     pub async fn shutdown(&mut self) -> anyhow::Result<()> {
-        Ok(self.child.kill().await?)
+        #[cfg(target_family = "unix")]
+        if self.prevent_zombie_on_unix_shutdown {
+            return self.shutdown_prevent_zombie().await;
+        }
```

Comment:
Is the flag really necessary? Seems we could just always do this since waiting on the pid is mostly harmless and avoid the clutter of piping the flag in a bunch of places. Makes it easier to remove later too if that happens.

Reply:
I am trying to localize the problem as much as possible. Temporalite doesn't need it, Windows doesn't need it. If you want me to use it for non-Windows-Temporalite though, I can.

Reply:
I am going to try to do this with Temporalite too and switch to SIGINT to have a more graceful shutdown

Reply:
Fixed